### PR TITLE
Add CarouselCreativeData

### PR DESCRIPTION
**Pull request by allentsai**
This commit adds a new object for FacebookCarousel to let us adjust certain
user settings.

PL-20580

### CODE REVIEW

**Comment by zhiachong** [comment]
👍

File: src/PaperG/FirehoundBlob/Facebook/Creative/FacebookCarouselCreativeData.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+use PaperG\FirehoundBlob\Facebook\FacebookCreativeData;
+use PaperG\FirehoundBlob\Utility;
+
+class FacebookCarouselCreativeData extends FacebookCreativeData
+{
+    use Utility;
+
+    const PRIMARY = "primary";
+    const MULTI_SHARE_END_CARD = "multiShareEndCard";
+    const MULTI_SHARE_OPTIMIZED = "multiShareOptimized";
```
Comment (allentsai) [inline src/PaperG/FirehoundBlob/Facebook/Creative/FacebookCarouselCreativeData.php]: enum, when we have time.  Would be a new version though, and break all the things.

File: src/PaperG/FirehoundBlob/Facebook/Creative/FacebookCarouselCreativeData.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+     */
+    public function getOptimized()
+    {
+        return $this->optimized;
+    }
+
+    /**
+     * @param mixed $primary
+     */
+    public function setPrimary($primary)
```
Comment (a37wong) [inline src/PaperG/FirehoundBlob/Facebook/Creative/FacebookCarouselCreativeData.php]: Maybe call it setPrimaryCard ?

File: src/PaperG/FirehoundBlob/Facebook/Creative/FacebookCarouselCreativeData.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        $this->primary = $primary;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getPrimary()
```
Comment (a37wong) [inline src/PaperG/FirehoundBlob/Facebook/Creative/FacebookCarouselCreativeData.php]: Offline discussion: isPrimary?

File: src/PaperG/FirehoundBlob/Facebook/Creative/FacebookCarouselCreativeData.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+     */
+    public function setEndCard($endCard)
+    {
+        $this->endCard = $endCard;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getEndCard()
```
Comment (a37wong) [inline src/PaperG/FirehoundBlob/Facebook/Creative/FacebookCarouselCreativeData.php]: isEndCard?

File: src/PaperG/FirehoundBlob/Facebook/Creative/FacebookCarouselCreativeData.php [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    /**
+     * @return mixed
+     */
+    public function getOptimized()
+    {
+        return $this->optimized;
+    }
+
+    /**
+     * @param mixed $primary
```
Comment (a37wong) [inline src/PaperG/FirehoundBlob/Facebook/Creative/FacebookCarouselCreativeData.php]: I assume the params are bools for both primary and endCard

File: src/PaperG/FirehoundBlob/Facebook/Creative/FacebookCarouselCreativeData.php [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+    public function __construct($array = null)
+    {
+        $this->fromArray($array);
+    }
+
+    /**
+     * @param mixed $endCard
+     */
+    public function setEndCard($endCard)
```
Comment (a37wong) [inline src/PaperG/FirehoundBlob/Facebook/Creative/FacebookCarouselCreativeData.php]: setUseEndCard

**Comment by a37wong** [comment]
:+1:

**Pull request merged**